opentelemetry/src/sdk/trace/tracer.rs

```diff
@@ -192,11 +190,13 @@ impl crate::trace::Tracer for Tracer {
                 _ => cx,
             }
         };
+        let span = parent_cx.span();
         let parent_span_context = if parent_cx.has_active_span() {
-            Some(parent_cx.span().span_context())
+            Some(span.span_context())
```

Comment:
The order seemed odd to me here. How can we get a span before checking if there is one? After looking at the implementation it makes sense now.

Out of curiosity: I think the following isn't the same, is it? Because I could write something like `Context::current_with_span(NoopSpan::new())`, in which case we should use explicitly provided invalid parent context.

```rust
let span = parent_cx.span();
let span.span_context().is_valid() {
    Some(span.span_context())
} else {
    None
}
```

Reply:
could check the spec but I think that is allowed? would be a way to make a new root span I think?

Reply:
Yeah, I think so. Oh well 😄